biogrid/apps/biogrid-mvp/src/app/pages/simulate-page/simulate-page.tsx

```diff
@@ -67,15 +67,20 @@ export const SimulatePage = () => {
     });
   };
 
-  const playSimulation = () => {
+  const playSimulation = async () => {
     const simResultsStateLen = simulationResults?.states.length || 0;
     if (currentStateFrame >= simResultsStateLen) {
       return;
     }
-    setTimeout(() => {
-      setCurrentStateFrame(currentStateFrame + 1);
-      playSimulation();
-    }, 1000);
+    const simResultsLen = simulationResults?.states.length || 0;
+    for (let i = 0; i < simResultsStateLen; i++) {
+      await new Promise((resolve, reject) => {
```

Comment:
why does this have to be nested in a promise?

Reply:
That is so that the for loop can 'sleep' between each cycle. This section is also changed in #87

Reply:
if it's changed in #87 and its not too much work, can we have this changed immeidately. Don't want to introduce this to master if we're immediately changing it anyway

Reply:
It's a relatively structural change in 87 to incorporate the pause. The new promise await is still there though as it is necessary to iteratevly go through a loop with sleep